backend/auth.py

```diff
@@ -14,35 +14,44 @@
 
 
 def get_token_auth_header(header):
-    """Obtains the Access Token from the Authorization Header
+    """Obtains the token from a given header
     """
-    auth = request.headers.get(header, None)
-    if not auth:
-        return Response(json.dumps({'error': 'Authorization_header_missing'}), 401)
 
-    parts = auth.split()
+    headerString = ""
+    try:
+        headerString = request.headers.get(header, None)
+    except HTTPError:
+        raise HTTPError("Cannot parse header")
 
-    if parts[0].lower() != 'bearer':
-        return Response("{'error': 'Invalid_header'}", 401)
+    if headerString is None:
+        raise TypeError("Header is missing")
 
-    elif len(parts) == 1:
-        return Response("{'error': 'Invalid_header'}", 401)
+    parts = headerString.split()
 
-    elif len(parts) > 2:
-        return Response("{'error': 'Invalid_header'}", 401)
+    if parts[0].lower() != 'bearer' or len(parts) == 1 or len(parts) > 2:
+        raise TypeError("Header invalid")
 
     token = parts[1]
     return token
 
 
+def eval_access_token():
```

Comment:
getAccessToken

Reply:
Since the function evaluates whether or not to actually use the access token, is it not clearer with this name?

Reply:
I don't think it's that clear that it returns the accessToken -_(*-*)_-

Reply:
Ye, did it yo way!